pyproject.toml

```diff
@@ -6,9 +6,13 @@ authors = ["The Aramis Lab"]
 license = "CC-BY-SA-4.0"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = ">=3.9,<3.11"
 jupyter-book = "^0.13.0"
-nibabel = "^2.3.3"
+numpy = ">=1.23"
+nibabel = ">=4.0.0"
+nilearn = ">=0.9.1"
+scipy = ">1.6.1"
+matplotlib = ">=3.5.0"
```

Comment:
Since you are only using `nilearn`, is there a specific reason why you need to specify numpy et al.?

Reply:
You should specify all dependencies that are imported *explicitly*. So far, it's `nibabel` and `nilearn` afaik. The rest should be pulled transitively.

Reply:
This works for me:

```diff
 [tool.poetry.dependencies]
-python = ">=3.9,<3.11"
+python = ">=3.8,<3.11"
 jupyter-book = "^0.13.0"
 nibabel = "^4"
-numpy = ">=1.23"
-nilearn = ">=0.9.1"
-scipy = ">1.6.1"
-matplotlib = ">=3.5.0"
+nilearn = {version = "^0.9", extras = ["plotting"]}
```

Now that we get a transitive dependency on `scipy`, we get to experience pain here too.